File: c++/string/halvesAreAlike.cc
#include <string>

using namespace std;

/* leetcode issue 1704*/
class Solution {
public:
    string st = "aeiou";
    bool judge(char c) {
        return st.find(tolower(c)) != string::npos;
    }

    bool halvesAreAlike(string s) {
        int a = 0, b = 0;
        int i = 0, j = s.size()-1;

        for(; i <=j; ++i, --j) {
            if(judge(s[i]))
                ++a;
            if(judge(s[j]))
                ++b;
        }

        return a == b;
    }
};

File: python/string/halvesAreAlike.py
#!/usr/bin/env python3
#coding=utf-8

''' leetcode issue 1704 '''
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        a = ['a','e','i','o','u','A','E','I','O','U']
        m = []
        n = []
        for i in list(s)[:len(list(s))//2]:
            if i in a:
                m.append(i)
        for j in list(s)[len(list(s))//2:]:
            if j in a:
                n.append(i)
        return bool(len(m) == len(n))
